test_unstructured/test_utils.py

```diff
@@ -313,42 +314,40 @@ def test_validate_data_args():
         utils.validate_date_args(None)
 
 
-def test_validate_date_args_accepts_standard_formats():
-    assert utils.validate_date_args("1990-12-01")
-    assert utils.validate_date_args("2050-01-01T00:00:00")
-    assert utils.validate_date_args("2050-01-01+00:00:00")
-    assert utils.validate_date_args("2022-02-12T14:30:00-0500")
+@pytest.mark.parametrize(
+    "date", ["1990-12-01", "2050-01-01T00:00:00", "2050-01-01+00:00:00", "2022-02-12T14:30:00-0500"]
+)
+def test_validate_date_args_accepts_standard_formats(date):
+    assert utils.validate_date_args(date)
 
 
-def test_validate_date_args_raises_for_invalid_formats():
-    with pytest.raises(ValueError):
-        assert utils.validate_date_args(None)
-        assert utils.validate_date_args("not a date")
-        assert utils.validate_date_args("1990-12-33")
-        assert utils.validate_date_args("2022-02-12T14:30:00-2000")
+@pytest.mark.parametrize("date", [None, "not a date", "1990-12-33"])
+def test_validate_date_args_raises_for_invalid_formats(date):
+    pattern1 = re.compile(r"The argument.*?(?:is None).*")
+    pattern2 = re.compile(r"The argument.*?(?:does not satisfy the format: YYYY-MM-DD).*")
+    combined_pattern = re.compile(f"({pattern1.pattern}|{pattern2.pattern})")
+    with pytest.raises(ValueError, match=combined_pattern):
+        assert utils.validate_date_args(date)
```

Comment:
Okay, this is fine, but hard to read :)

Leave it this time but in future I'd separate this into two tests, one for each different type, which I expect must be in different places in the code.

Actually I think I'd just change the messages to start with something like "invalid date: ..." and then just inline the match like:
```python
with pytest.raises(ValueError, match="invalid date: "):
    ...
```

But like I say, good enough for present purposes. Just a little bit "cure worse than disease" :)

Also, remember your intent, which is to illuminate for the reader what happens, like "the user gets an error message like this". A complicated regex doesn't serve that intent and we don't really care that the message exactly matches some form, just do your best to give the reader an idea instead of leaving no match.